lib/rules/switch-final-break.js

```diff
@@ -0,0 +1,158 @@
+            switchFinalBreakAlways:
+                "Final clause in 'switch' statement should end with 'break;'."
+        },
+        schema: [
+            {
+                enum: ["never", "always"]
+            }
+        ],
+        type: "suggestion"
+    },
+    create(context) {
+        let currentCodePath = null;
+        const requireBreak = context.options[0] ? context.options[0] : "never";
+
+        /**
+         * Returns the statements list of a given `SwitchCase` node
+         * @param {ASTNode} node `SwitchCase` node
+         * @returns {ASTNode[]} - the statements list of the `SwitchCase` node
+         */
+        function getStatements(node) {
+            if (
+                node.consequent.length === 1 &&
+                node.consequent[0].type === "BlockStatement"
+            ) {
+                return node.consequent[0].body;
+            }
+            return node.consequent;
+        }
+
+        /**
+         * Returns the last statement of a given switch clause
+         * @param {ASTNode} node switch clause node
+         * @returns {ASTNode} - the final statement of the switch clause
+         */
+        function getLastStatement(node) {
+            return lodash.last(getStatements(node));
+        }
+
+        /**
+         * Checks whether or not a given code path segment is reachable.
+         * @param {CodePathSegment} segment A CodePathSegment to check.
+         * @returns {boolean} `true` if the segment is reachable.
+         */
+        function isReachable(segment) {
+            return segment.reachable;
+        }
+
+
+        return {
+            onCodePathStart(codePath) {
+                currentCodePath = codePath;
+            },
+            onCodePathEnd() {
+                currentCodePath = currentCodePath.upper;
+            },
+
+            "SwitchCase:exit"(node) {
+                if (requireBreak !== "always") {
+                    return;
+                }
+                if (node.consequent.length === 0) {
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            node,
+                            " break;"
+                        )
+                    });
+                }
+                if (currentCodePath.currentSegments.some(isReachable) &&
+                    node.consequent.length > 0 &&
+                    lodash.last(node.parent.cases) === node
+                ) {
+                    const lastStatement = getLastStatement(node);
+
+                    context.report({
+                        node,
+                        messageId: "switchFinalBreakAlways",
+                        fix: fixer => fixer.insertTextAfter(
+                            lastStatement,
+                            " break;"
+                        )
```

Comment:
This:

```js
/* eslint switch-final-break: ["error", "always" ] */

switch (foo) {
  case a:
    bar()
}
```

would be auto-fixed to:

```js
/* eslint switch-final-break: ["error", "always" ] */

switch (foo) {
  case a:
    bar() break;
}
```

which is `SyntaxError: Unexpected token 'break'`.

Maybe we should always insert a linebreak, e.g. `"\nbreak;"`?

Also, I'm not sure should there be `;` at the end, or just `"\nbreak"` (and let the `semi` rule insert the `;` if configured so).